### ISSUE TITLE
Some issues about Brain demo

### ISSUE THREAD

**Issue by Yoyo-cool99** [avdravid/medXGAN_explanations#1]
Hi, Your work is very meaningful and inspired me a lot. However, I have some issues about when I am running the Brain Demo.

When I'm reconstructing the example image, set the generator from train mode to evaluation mode,and the probability of reconstructed negative image is reduced([0.78,0.21]->[0.53,0.46]).I am very confused about this and have tried many times.

Would you mind giving me some suggestions about that?

Thanks a lot in advance!

<img width="242" alt="iShot2022-07-21_15 51 08" src="https://user-images.githubusercontent.com/50907124/180160166-a7445caa-a72d-4ac6-9b64-fc58208583a6.png">
<img width="242" alt="iShot2022-07-21_15 51 18" src="https://user-images.githubusercontent.com/50907124/180160093-bd2b71ef-9456-435b-a074-32802c94ea6e.png">

**Comment by avdravid** [comment]
Hi,
As an undergrad, it's very nice to hear this. I am glad you found the work
interesting.  When you change the generator from train mode to evaluation
mode in PyTorch, the internal representations of the model change because
you turn off BatchNorm layers. In this case, it seems that pixels may
slightly change (that we as humans can't really see), but which change the
classifier's confidence. Please let me know if you have any other
questions.

**Comment by Yoyo-cool99** [comment]
Thank you for your answer. 
But I found that in lung example where the generator is set to evaluation mode in the code, it works fine.
Is this due to the small size of the brain training set or other reasons?

**Comment by avdravid** [comment]
Hi,

It's hard to tell why it doesn't change that much for the brain example
since we would need to know the internal mechanisms of the neural networks.
Possibly. During evaluation, it uses the batch statistics from the training
distribution.

**Comment by Yoyo-cool99** [comment]
Okay, thanks again for your answer.
The last question is whether the medXGAN is suitable for explaining other classifiers, such as cat and dog binary classifier, face binary classifier. 
Despite the computational overhead of training a GAN, I want to do some experiments on these classifiers.

**Comment by avdravid** [comment]
No problem. The medxgan exploits the domain knowledge of underlying
structure and overlying class features present in medical images. So I
wouldn’t imagine it working well with the cases you mentioned. I would
suggest looking at this paper:
https://explaining-in-style.github.io. Thanks again for your interest!
